plugins/saveactions.c

```diff
@@ -260,23 +262,36 @@ static void instantsave_document_new_cb(GObject *obj, GeanyDocument *doc, gpoint
 {
     if (enable_instantsave && doc->file_name == NULL)
     {
+		const gchar *directory;
 		gchar *new_filename;
 		gint fd;
 		GeanyFiletype *ft = doc->file_type;
 
-		fd = g_file_open_tmp("gis_XXXXXX", &new_filename, NULL);
-		if (fd != -1)
-			close(fd); /* close the returned file descriptor as we only need the filename */
-
 		if (ft == NULL || ft->id == GEANY_FILETYPES_NONE)
 			/* ft is NULL when a new file without template was opened, so use the
 			 * configured default file type */
 			ft = filetypes_lookup_by_name(instantsave_default_ft);
 
-		if (ft != NULL)
-			/* add the filetype's default extension to the new filename */
+		/* construct filename */
+		directory = !EMPTY(instantsave_target_dir) ? instantsave_target_dir : g_get_tmp_dir();
+		new_filename = g_build_filename(directory, "gis_XXXXXX", NULL);
+		if (ft != NULL && !EMPTY(ft->extension))
 			SETPTR(new_filename, g_strconcat(new_filename, ".", ft->extension, NULL));
```

Comment:
`ft == NULL` means `doc->file_type` is NULL, but its not set before being dereferenced at **here** below

Reply:
Great, you are uncovering very old bugs :). Should be fixed.